.github/mkosi.extra/usr/lib/systemd/system/mkosi-check-and-shutdown.service

```diff
@@ -0,0 +1,13 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+[Unit]
+Description=Check if any service failed and then shutdown the machine
+After=multi-user.target network-online.target
+Requires=multi-user.target
+OnFailure=poweroff.target
+OnFailureJobMode=replace-irreversibly
+
+[Service]
+Type=oneshot
+ExecStartPre=-rm -f /failed-services /testok
```

Comment:
`-` is not useful, I think. If the removal fails, we probably want to abort the test.